strapi/strapi
# [TS] Automatic types generation (dev mode)

### What does it do?

> :warning: Experimental feature, disabled by default.
To enable it, set the `typescript.autogenerate` config property to true

- Automatically generate the schema.d.ts file on server restart (only when using the develop command)
- Adds a configuration option: `typescript.autogenerate` to enable or disable the automatic generation

Note: This PR only aims to introduce the base functionality that will evolve over time when we'll implement a new type system (services, type augmentation, etc...)

### Why is it needed?

Improve the DX by matching the generated types with the current app structure

### How to test it?

- set the `typescript.autogenerate` config property to true
- run `yarn develop` in the getstarted, it should start as fast as before
- once the app started, take a look at your app folder, you should see a newly generated `schemas.d.ts` file that contains type definitions for your application
- try to create a new content type (or component) or modify an existing one, upon server restart the type definitions should have been updated to reflect the changes
